egs/wsj/s5/steps/dur_model/make_nnet_config.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+learning_rate=0.001
+natural_gradient=false
+hidden_dim1=
+hidden_dim2=
+
+[ -f ./path.sh ] && . ./path.sh; # source the path.
+. parse_options.sh || exit 1;
+
+if [ $# != 2 ]; then
+   echo "Usage: $0 [options] <input-dim> <output-dim>"
+   echo "e.g.: $0 320 50"
+   exit 1;
+fi
+
+feat_dim=$1
+output_dim=$2
+
+if [ -z $hidden_dim1 ]; then
+  hidden_dim1=$[$feat_dim*3]
+fi
+
+if [ -z $hidden_dim2 ]; then
+  hidden_dim2=$[$output_dim*2]
+fi
+
```

Comment:
So when you had the bottleneck, was it as an extra layer, or you just
reduced the size of the second hidden layer?  Probably an extra small layer
would work better than reducing the size of the second hidden layer; and
you might want to remove the ReLU after it.

On Sun, Jan 17, 2016 at 3:00 PM, Hossein Hadian <notifications@github.com>
wrote:

> In egs/wsj/s5/steps/dur_model/make_nnet_config.sh
> <https://github.com/kaldi-asr/kaldi/pull/414#discussion_r49951290>:
>
> > +   echo "Usage: $0 [options] <input-dim> <output-dim>"
> > +   echo "e.g.: $0 320 50"
> > +   exit 1;
> > +fi
> > +
> > +feat_dim=$1
> > +output_dim=$2
> > +
> > +if [ -z $hidden_dim1 ]; then
> > +  hidden_dim1=$[$feat_dim*3]
> > +fi
> > +
> > +if [ -z $hidden_dim2 ]; then
> > +  hidden_dim2=$[$output_dim*2]
> > +fi
> > +
>
> Yes. In my first experiments I used the bottleneck network (dim = 10) but
> the results were not very good. I have not experimented it with all other
> options (max-duration, hidden-layer1 dimension, context size) yet. Actually
> I am currently testing it with a good context size, but the results are not
> yet ready.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/414/files#r49951290>.
>